### ISSUE TITLE
Not possible to combine an ngf-drop and ngf-select with same model

### ISSUE THREAD

**Issue by intellix** [danialfarid/ng-file-upload#1469]
**Angular**: 1.5.3
**ngFileUpload**: 12.0.4

So I've been fighting with this for quite a number of hours now and still haven't been able to achieve it. For a better user experience, I'm trying to have the ngf-drop as high up the page as I can, so they don't have to put it into a small box. I've also got an ngf-select so the user can use the button to select files if they wish.

In it's simplified form, it looks like:

```html
<form>
  <div class="drag-area" ngf-drag-over-class="'drag-over'" name="files" ng-model="app.files" ngf-drop="app.drop(myForm)" ngf-multiple="true" required>
    <p>Drag in a file or select below to upload</p>
    <button class="btn btn-block btn-info" type="file" name="files" ng-model="app.files" ngf-select="app.submit(myForm)" ngf-multiple="true" required>Select file(s)</button>
  </div>
</form>
```

I've also tried having 2 select fields in my form to test what happens when you duplicate a control with the same name/model and see how it validates upon selecting a value from one. It creates 2x form controls but they're both valid upon selecting one.

The problem with the ngf-drop and ngf-select using the same name/model is that only one of them is validated upon use. This means the user has to drop a file and then select a file.

Been debugging a little to figure out when things are getting validated as the library is performing a bit of magic. I see that if I specify an ngf-select then it creates a hidden `input[type="file"]` but not on an ngf-drop.

Codepen: http://codepen.io/intellix/pen/JXpJrQ?editors=1010

**Comment by intellix** [comment]
Investigating this a little more. In my example, if I change the <select> field, the Angular required validator is run twice (for both fields) and they both return that they're valid.

With the file-drop functionality, despite there being 2x fields with the same name, the required validator is only run once (I suppose for the dropper and not being run for the <button ngf-select>.

**Comment by danialfarid** [comment]
What the issue is here?

**Comment by intellix** [comment]
TL;DR - you can't have a `<div ngf-drop>` and `<button ngf-select>` with the same name/model and have a valid form if you use either. Only one of the items is valid.
I've got around this by naming them different and doing this for validity check:

```html
<div name="fileDrop" ng-model="documents.files" ngf-drop="documents.submit(documentForm)" required>
  <p>Drop file here to upload or</p>
  <button type="button" class="btn" name="files" ng-model="documents.files" ngf-select="documents.submit(documentForm)" required>Select files</button>
</div>
```

```js
if (form.type.$valid && (form.files.$valid || form.fileDrop.$valid)) {
   // valid
}
```

**Comment by danialfarid** [comment]
That workaround seems good enough

**Comment by CareerFairPlus** [comment]
Having trouble doing this with angular material and using ng-messages to display error messages based upon what is invalid about the form.  When using two models, only one is ever valid, so ng-messages always shows an error for the other model.
